### PR TITLE
remove letters from money mask

### PR DESCRIPTION
**Pull request by PH7-Jack**
This PR solves the problem of adding letters on empty inputs using the money mask, the mask stops working

**Unexpected Behavior:**
Ability to type letters on money input
image

**Expected Behavior:**
Letters and non-numeric characters should be removed

**Live example:**
1. Go to the alpine mask (documentation)[https://alpinejs.dev/plugins/mask#money-inputs]
2. Click in the money input
3. Type **"ABC"**

### CODE REVIEW

File: packages/mask/src/index.js [hunk 1]
```diff
@@ -164,6 +164,8 @@ export function buildUp(template, input) {
 }
 
 function formatMoney(input, delimeter = '.', thousands) {
+    if (!input.replace(/\D+/g, '')) return '9'
```
Comment (ekwoka) [inline packages/mask/src/index.js:167]: Shouldn't this be a regex test and not a replace?

**Review by ekwoka — CHANGES_REQUESTED** [review]

File: packages/mask/src/index.js (hunk 1 above)
Comment (PH7-Jack) [reply packages/mask/src/index.js:167]: @ekwoka Yes, it is possible to use the both

Comment (ekwoka) [reply packages/mask/src/index.js:167]: I don't mean whats possible, I mean what's the correct tool for the job?

Wouldn't

```js
if (!/^\D+$/.test(input)) return '9'
```

Be more explicit to what you're actually doing?

**Comment by joshhanley** [comment]
@PH7-Jack I have merged in main and there are some failing tests. Could you please address them and also change to using `.test()` as suggested by @ekwoka? Thanks!

**Comment by joshhanley** [comment]
@PH7-Jack turns out the failing test is unrelated, I've pushed a fix to main, so leave it with me for the moment and I will merge in changes and check it's all good.

**Comment by PH7-Jack** [comment]
@joshhanley Ok, if you need something just let me know.
Thanks

**Comment by joshhanley** [comment]
@PH7-Jack thanks for the PR! I've updated it to use regex test and merged it.

**Pull request merged**